core/sdk-core/src/main/java/software/amazon/awssdk/core/internal/http/pipeline/stages/MakeHttpRequestStage.java

```diff
@@ -53,8 +63,20 @@ public MakeHttpRequestStage(HttpClientDependencies dependencies) {
 
     private SdkHttpFullResponse executeHttpRequest(SdkHttpFullRequest request, RequestExecutionContext context) throws Exception {
         final AbortableCallable<SdkHttpFullResponse> requestCallable = sdkHttpClient
-                .prepareRequest(request, SdkRequestContext.builder().build());
+            .prepareRequest(request, SdkRequestContext.builder().build());
+
+        long timeoutInMillis = resolveTimeoutInMillis(context.requestConfig()::apiCallAttemptTimeout, apiCallAttemptTimeout);
+
+        TimeoutTracker timeoutTracker = timeSyncTaskIfNeeded(timeoutExecutor, timeoutInMillis, false);
+        timeoutTracker.abortable(requestCallable);
+
+        context.apiCallAttemptTimeoutTracker(timeoutTracker);
+        context.apiCallTimeoutTracker().abortable(requestCallable);
 
-        return requestCallable.call();
+        try {
+            return requestCallable.call();
```

Comment:
We don't currently do this in V2, but I guess we can add a `isBufferResponse` in `ExecuteRequest` and pass it to the http clients. This seems a bit out of scope of this PR and I'd say creating another PR for this feature.

Reply:
One of the goals of the new timers is to make the per attempt and per call timeouts behave more similiarly. So the per attempt timeout would not just be an abort on the HTTP request but a timer that interrupts as well and is scoped to the entirety of the things we do per request (i.e. unmarshalling for example). This would elimnate the need for buffering as we consume the data in unmarshalling which is subject to the timeout. Is this how the timeouts work (haven't looked at the PR)?

Reply:
No, the current timeout implementation is the same as v1, so attempt timeout is just aborting the request. I like the idea of scoping it to the entire thing and will update the PR to see how it looks. One concern I have though is now that both call timemout and attempt timeout can interrupt the thread, handling interrupt exception can get tricky and we might get more issues with uncleared interrupt status.